File: 1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */
func twoSum(nums []int, target int) []int {
	cache := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		last := target - nums[i]
		if j, ok := cache[last]; ok {
			return []int{i, j}
		}
		cache[nums[i]] = i
	}
	return nil
}

File: 0001.two-sum/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) < 2:
            return []
        cache = {}
        for i in range(0,len(nums)):
            last = target - nums[i]
            if cache.__contains__(last):
                return [i, cache[last]]
            cache[nums[i]]=i
        return []


# @lc code=end
